mcpserver/tools/file_utils.go

```diff
@@ -13,17 +13,23 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/mattermost/mattermost-plugin-ai/mcpserver/types"
 	"github.com/mattermost/mattermost/server/public/model"
 )
 
-// fetchFileData fetches file data from a file path or URL and returns it as []byte
-func fetchFileData(filespec string) ([]byte, error) {
+// fetchFileDataForLocal fetches file data from a file path or URL (local access only)
+func fetchFileDataForLocal(filespec string, accessMode types.AccessMode) ([]byte, error) {
 	if filespec == "" {
 		return nil, fmt.Errorf("empty filespec provided")
 	}
 
+	// URLs are only allowed for local access mode
+	if accessMode != types.AccessModeLocal {
```

Comment:
I also added an accessMode check for each of these functions just to ensure they aren't used in the wrong place in future.